### ISSUE TITLE
Huge vendors~main.js after building extensions

### ISSUE THREAD

**Issue by maresb** [jupyterlab/jupyterlab#6869]
I'm primarily running JupyterLab 1.0.1 on Docker, but I have also observed similar behavior with 1.0.2 in regular Anaconda / Ubuntu 18.04.  

After getting my extensions configured, I noticed that it was taking a ridiculously long time for JupyterLab to load.  Upon inspecting the page load, I noticed that the `vendors~main.js` file was over 70MB, and this was the cause of the slowdown.  Looking into the file, I observed that **it had not been minified**.

Being in Docker, I can easily do experiments in various environments.  For instance, I discovered that just by running `jupyter lab build`, without even installing any extensions, my vendors~main.js grows from 4MB to 28MB:

Running [jupyter/base-notebook:7a3e968dd212 (JupyterLab 1.0.1)](https://github.com/jupyter/docker-stacks/blob/7a3e968dd21268c4b7a6746458ac34e5c3fc17b9/base-notebook/Dockerfile): 

```
jovyan@c0945309f77d:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors*
-rw-rw-rw- 1 jovyan users  4174751 Jul  3 00:23 /opt/conda/share/jupyter/lab/static/vendors~main.268f77d7204487100f7e.js
-rw-rw-rw- 1 jovyan users 13422562 Jul  3 00:23 /opt/conda/share/jupyter/lab/static/vendors~main.268f77d7204487100f7e.js.map
jovyan@c0945309f77d:~$ jupyter lab build
[LabBuildApp] JupyterLab 1.0.1
[LabBuildApp] Building in /opt/conda/share/jupyter/lab
[LabBuildApp] Building jupyterlab assets
jovyan@c0945309f77d:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors*
-rw-r--r-- 1 jovyan users 28696849 Jul 22 17:26 /opt/conda/share/jupyter/lab/static/vendors~main.34584ab9ba8ddf7bdf5d.js
```
If I install some extensions it's even worse, jumping to 74MB

```
jovyan@c01fb50160d0:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors*
-rw-rw-rw- 1 jovyan users  4174751 Jul  3 00:23 /opt/conda/share/jupyter/lab/static/vendors~main.268f77d7204487100f7e.js
-rw-rw-rw- 1 jovyan users 13422562 Jul  3 00:23 /opt/conda/share/jupyter/lab/static/vendors~main.268f77d7204487100f7e.js.map
jovyan@2bed5d05c292:~$ jupyter labextension install @jupyter-widgets/jupyterlab-manager@1.0.1 --no-build \
>    && jupyter labextension install plotlywidget@1.0.0 --no-build \
>    && jupyter labextension install jupyterlab-plotly@1.0.0 --no-build \
>    && jupyter labextension install jupyterlab-chart-editor@1.2 --no-build \
>    && jupyter labextension install jupyter-matplotlib@0.4.2 --no-build \
>    && jupyter lab build
[LabBuildApp] JupyterLab 1.0.1
[LabBuildApp] Building in /opt/conda/share/jupyter/lab
[LabBuildApp] Building jupyterlab assets
jovyan@2bed5d05c292:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors~main*
-rw-r--r-- 1 jovyan users 74224009 Jul 22 18:00 /opt/conda/share/jupyter/lab/static/vendors~main.0b7dffccd348c81c9668.js
```
Comparing with [jupyter/base-notebook:38f518466042 (JupyterLab 0.35.5)](https://github.com/jupyter/docker-stacks/blob/38f518466042796d4c015e47778a7b6c94ad6905/base-notebook/Dockerfile), the vendors~main.js file starts as 8.5MB and increases only a negligible amount after running `jupyter lab build`.

```
jovyan@bed496257af4:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors*
-rw-rw-r-- 1 jovyan users 8483812 May 12 15:15 /opt/conda/share/jupyter/lab/static/vendors~main.bc0f18ca5057e1affd13.js
-rw-rw-r-- 1 jovyan users 9749471 May 12 15:15 /opt/conda/share/jupyter/lab/static/vendors~main.bc0f18ca5057e1affd13.js.map
-rw-rw-r-- 1 jovyan users 3117448 May 12 15:15 /opt/conda/share/jupyter/lab/static/vendors~vega.6abdc5f29e610075ab37.js
-rw-rw-r-- 1 jovyan users 2426181 May 12 15:15 /opt/conda/share/jupyter/lab/static/vendors~vega.6abdc5f29e610075ab37.js.map
jovyan@bed496257af4:~$ jupyter lab build
[LabBuildApp] JupyterLab 0.35.5
[LabBuildApp] Cleaning /opt/conda/share/jupyter/lab
Cleaning /opt/conda/share/jupyter/lab...
Success!
[LabBuildApp] Building in /opt/conda/share/jupyter/lab
[LabBuildApp] Node v11.14.0

<Cut out several pages of verbose logs.  I can provide them on request.>

Done in 10.62s.
jovyan@bed496257af4:~$ ls -l $CONDA_DIR/share/jupyter/lab/static/vendors*
-rw-r--r-- 1 jovyan users 8500582 Jul 22 17:33 /opt/conda/share/jupyter/lab/static/vendors~main.e4b2f4a67b6ba52e5f57.js
-rw-r--r-- 1 jovyan users 9755081 Jul 22 17:33 /opt/conda/share/jupyter/lab/static/vendors~main.e4b2f4a67b6ba52e5f57.js.map
-rw-r--r-- 1 jovyan users 2964334 Jul 22 17:33 /opt/conda/share/jupyter/lab/static/vendors~vega.a699d0cffd013e4c68bd.js
-rw-r--r-- 1 jovyan users 2453665 Jul 22 17:33 /opt/conda/share/jupyter/lab/static/vendors~vega.a699d0cffd013e4c68bd.js.map
```
I don't know much about nodejs development.  Could the lack of minification account for the huge increase in size and consequently long loading times?  Please let me know if you have any suggestions about how to proceed and how to reduce the size of vendors~main.js.  Thanks!

**Comment by telamonian** [comment]
Duplicate of #6804. This'll get fixed in #6839

**Comment by maresb** [comment]
I look forward to the release of  JL 1.1.  In the meantime, I solved my problem by simply running
```
jupyter lab build --dev-build=False
```
Thanks for the references!
